src/controllers/csi/provisioner/controller.go

```diff
@@ -76,6 +79,27 @@ func (provisioner *OneAgentProvisioner) SetupWithManager(mgr ctrl.Manager) error
 }
 
 func (provisioner *OneAgentProvisioner) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	fs := afero.NewOsFs()
+	if provisioner.maxParallel > 0 {
+		if provisioner.currentParallel == provisioner.maxParallel {
+			log.Info("max parallel downloads reached, requeued")
+			return reconcile.Result{RequeueAfter: shortRequeueDuration}, nil
+		}
+		atomic.AddInt64(&provisioner.currentParallel, 1)
+		log.Info("staring parallel download")
+		go func() {
+			_, err := provisioner.reconcile(ctx, request, fs)
+			if err != nil {
+				log.Error(err, "Problem while provisioning oneagents in parallel")
+			}
+			atomic.AddInt64(&provisioner.currentParallel, -1)
+		}()
```

Comment:
if `provisioner.reconcile()` panicked, `currentParallel` wouldn't decrease :boom:.
```suggestion
		go func() {
			defer atomic.AddInt64(&provisioner.currentParallel, -1)
			_, err := provisioner.reconcile(ctx, request, fs)
			if err != nil {
				log.Error(err, "Problem while provisioning oneagents in parallel")
			}
		}()
```